### PR TITLE
update installation doc

### PR DESCRIPTION
**Pull request by dalg24**
doc on how to install Trilinos would be nice
reference to Dockerfile for cap-stack is another option

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -54,6 +54,10 @@ Cap will search for the following libraries at configuration-time:
 
     * Boost.MPI
 
+    * Boost.Python (if you enabled Cap with Python)
```
Comment (Rombur) [inline docs/source/install.rst]: Should add iostream, system, and thread. Otherwise, deal.II could use a different version of boost

**Comment by dalg24** [comment]
I thought about it. But then we'd have to dig into Trilinos and do same thing...

File: docs/source/install.rst [hunk 2]
```diff
@@ -112,6 +110,18 @@ Then run:
     $ ../configure_dealii
     $ make -j<N> install
 
+Please refer to the deal.II documentation to see how to install p4est and
+Trilinos.
+
+Here is how you can build p4est:
```
Comment (Rombur) [inline docs/source/install.rst]: I don't think this is necessary. deal has a script to install p4est. Just send them to deal.

**Comment by Rombur** [comment]
Then just say that you need everything all the time.

**Pull request merged**